core/services/directrequestocr/drlistener.go

```diff
@@ -313,14 +314,57 @@ func (l *DRListener) setError(ctx context.Context, requestId RequestID, runId in
 	}
 }
 
+func (l *DRListener) parseRequestData(request *ocr2dr_oracle.OCR2DROracleOracleRequest) pipeline.FinalResult {
```

Comment:
Did you add this only to be able to extract aggregation data? If so then a better way is to read it from DB just like we read parsed result/error at the end (see below).

I initially thought you moved the whole pipeline definition here, which would be an interesting idea. Keeping it split between here and job spec doesn't make sense. In any case, we want to eventually get rid of the pipeline altogether (https://smartcontract-it.atlassian.net/browse/FUN-135)

Reply:
Yeah, added it to parse the aggregation method from the request data so that I can store it in the DB to access later in aggregation.

Do you mean here?
```
	computationResult, errResult = ExtractRawBytes(computationResult)
```
The problem was that this only returns the "result" bytes, not the aggregation method.
I think based on your other suggestion of including it in the observation I could instead not even store it in the DB, and instead just mode it from the observations.

Reply:
I meant here:
`_, cborParseErr := l.jobORM.FindTaskResultByRunIDAndTaskName(run.ID, CBORParseTaskName, pg.WithParentCtx(ctx))`

The return value that we currently ignore has the full JSON that user sent as "data". You can extract aggMethod field from it.

Observation phase reads stuff from the DB to you need to store it in the DB no matter what. DB is the only way to communicate between Listener and OCR plugin.